packages/compass-shell/src/modules/worker-runtime/worker.js

```diff
@@ -0,0 +1,64 @@
+import { parentPort } from 'worker_threads';
+import { ElectronRuntime } from '@mongosh/browser-runtime-electron';
+import { CliServiceProvider } from '@mongosh/service-provider-server';
+import { exposeAll, createCaller } from './rpc';
+import { serializeResult, serializeError } from './worker-shell-result';
+
+let provider = null;
+let runtime = null;
+
+const evaluationListener = createCaller(
+  ['onPrint', 'onPrompt', 'toggleTelemetry'],
+  parentPort
+);
+
+async function init(uri, options, cliOptions) {
+  provider = await CliServiceProvider.connect(uri, options, cliOptions);
+  runtime = new ElectronRuntime(provider /** , TODO: messageBus? */);
+
+  runtime.setEvaluationListener(evaluationListener);
+
+  return true;
+}
+
+async function evaluate(code) {
+  try {
+    if (!runtime) {
+      throw new Error('Runtime is not initiated yet');
+    }
+
+    const result = await runtime.evaluate(code);
```

Comment:
We can cancel sync op with `SIGINT` now, but async op is not handled in the worker. Something similar to [async-repl implementation](https://github.com/mongodb-js/mongosh/blob/4ac387f983e6973e54bbcbe494f700438f2c2ec6/packages/cli-repl/src/async-repl.ts#L40-L53) should be added there, but I think doesn't need to be implemented int he scope of the PoC https://github.com/mongodb-js/mongosh/blob/4ac387f983e6973e54bbcbe494f700438f2c2ec6/packages/cli-repl/src/async-repl.ts#L40-L53